src/pal/src/synchmgr/synchcontrollers.cpp

```diff
@@ -578,21 +578,22 @@ namespace CorUnix
     waiting threads awakening.
     --*/
     PAL_ERROR CSynchStateController::IncrementSignalCount(
-        DWORD dwAmountToIncrement)
+        LONG lAmountToIncrement)
     {
         VALIDATEOBJECT(m_psdSynchData);
 
-        LONG lOldCount = m_psdSynchData->GetSignalCount();        
-        LONG lNewCount = lOldCount + (LONG)dwAmountToIncrement;
-
         _ASSERTE(InternalGetCurrentThread() == m_pthrOwner);
-        _ASSERT_MSG((DWORD)INT_MAX > dwAmountToIncrement,
+        _ASSERTE(lAmountToIncrement > 0);
```

Comment:
I assume that you've checked all callers and having lAmountToIncrement to be 0 is indeed illegal?